### PR TITLE
Week03 submissions

### PR DESCRIPTION
**Pull request by Evan-CSE**

### CODE REVIEW

File: week03/design_hashmap.cpp [hunk 1]
```diff
@@ -0,0 +1,53 @@
+// TC: O(1)
+// MC: O(n) where n is the number of operations
+
+class MyHashMap
+{
+public:
+    const static int MAXSIZE = 103880;
```
Comment (Sadman007) [inline week03/design_hashmap.cpp:7]: too large.

File: week03/design_hashmap.cpp [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+            if ((*it).first == key)
+                return (*it).second;
+        }
+        return -1;
+    }
+
+    void remove(int key)
+    {
+        key = get_hash(key);
+        for (auto it = Map[key].begin(); it != Map[key].end(); it++)
```
Comment (Sadman007) [inline week03/design_hashmap.cpp:45]: Why not `for(auto it : Map)`?

File: week03/group_shifted_string.cpp [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+        map<string, vector<string>> same_tp_str_vc;
+        string upper_string;
+        int distance;
+        vector<vector<string>> grp_of_str;
+        for (auto &x : strings)
+        {
+            upper_string = "";
+            distance = 'z' - x[0];
+            for (auto &ch : x)
+                upper_string += (ch - 'a' + distance) % 26 + 'a';
```
Comment (Sadman007) [inline week03/group_shifted_string.cpp:21]: Why didn’t you create another function that calculates the hash value.

File: week03/isomorphic_string.cpp [hunk 4]
```diff
@@ -0,0 +1,28 @@
+// TC:O(n) where n denotes size of the given string
+// MC:O(1)
+class Solution
+{
+public:
+    bool isIsomorphic(string s, string t)
+    {
+        unordered_map<char, char> rplc_with;
+        int parent[400];
```
Comment (Sadman007) [inline week03/isomorphic_string.cpp:9]: Extremely bad! Always use vector in this case.

**Review by Sadman007 — APPROVED** [review]

File: week03/design_hashmap.cpp (hunk 1 above)
Comment (Evan-CSE) [reply week03/design_hashmap.cpp:7]: Vai, will it be a problem in interview round?

File: week03/design_hashmap.cpp (hunk 2 above)
Comment (Evan-CSE) [reply week03/design_hashmap.cpp:45]: I was facing some problem ; so to ensure if everything's find I used this method.

**Pull request merged**